### PR TITLE
C wrapper for linsolve and solve_poly

### PR DESCRIPTION
**Pull request by ahumenberger**

### CODE REVIEW

**Comment by ShikharJ** [comment]
Hi @ahumenberger. You can just run `./bin/test_format_local.sh` before committing, that would automatically take care of the formatting.

File: symengine/cwrapper.h [hunk 1]
```diff
@@ -611,6 +611,15 @@ CWRAPPER_OUTPUT_TYPE function_symbol_set(basic s, const char *c,
 CWRAPPER_OUTPUT_TYPE basic_coeff(basic c, const basic b, const basic x,
                                  const basic n);
 
+//! Wrapper for solve.h
+
+//! Solves the system of linear equations given by sys
+CWRAPPER_OUTPUT_TYPE vecbasic_linsolve(const CVecBasic *sys,
+                                       const CVecBasic *sym, CVecBasic *sol);
```
Comment (isuruf) [inline symengine/cwrapper.h]: Usually we have the output as the first argument.

Comment (ahumenberger) [reply symengine/cwrapper.h]: OK, thanks for the info. But there is certainly some inconsistency:
`basic_free_symbols(const basic self, CSetBasic *symbols)`

However, the output should be the first argument, and there are no exceptions to that?!

Comment (isuruf) [reply symengine/cwrapper.h]: Yes, I'd like to change `basic_free_symbols` and `basic_get_args` too. Don't know how we missed those.

**Comment by isuruf** [comment]
Thanks

**Pull request merged**